### PR TITLE
Daravind/more exe tests 2. Ported a bunch of tests from codeplex

### PR DESCRIPTION
**Pull request by deepakaravindr**
Daravind/more exe tests 2. Ported a bunch of tests from codeplex.
This PR has 5 commits and only test code has been changed

@yishaigalatzer @pranavkm @emgarten 

Filed a bunch of issues while making this change
- https://github.com/NuGet/Home/issues/1026
- https://github.com/NuGet/Home/issues/1029
- https://github.com/NuGet/Home/issues/1031
- https://github.com/NuGet/Home/issues/1050
- https://github.com/NuGet/Home/issues/1051
- https://github.com/NuGet/Home/issues/1052

### CODE REVIEW

File: test/NuGet.CommandLine.Test/NuGetConfigCommandTest.cs [hunk 1]
```diff
@@ -25,16 +24,25 @@ public void ConfigCommand_ChangeDefaultConfigFile()
                 };
 
                 // Act
-                int result = Program.Main(args);
+                // Set the working directory to C:\, otherwise,
+                // the test will change the nuget.config at the code repo's root directory
+                int result = Program.MainCore(@"c:\", args);
```
Comment (pranavkm) [inline test/NuGet.CommandLine.Test/NuGetConfigCommandTest.cs:29]: Can we figure out the current volume or something like this?

File: test/NuGet.CommandLine.Test/NuGetListCommandTest.cs [hunk 2]
```diff
@@ -0,0 +1,563 @@
+﻿using System;
```
Comment (pranavkm) [inline test/NuGet.CommandLine.Test/NuGetListCommandTest.cs:1]: License header.

File: test/NuGet.CommandLine.Test/NuGetListCommandTest.cs [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+                        string feed = server.ToODataFeed(new[] { package1, package2 }, "Search");
+                        MockServer.SetResponseContent(response, feed);
+                    }));
+                server.Get.Add("/nuget", r => "OK");
+
+                server.Start();
+
+                // Act
+                var args = "list test -IncludeDelisted -Source " + mockServerEndPoint + "nuget";
+                var r1 = CommandRunner.Run(
```
Comment (pranavkm) [inline test/NuGet.CommandLine.Test/NuGetListCommandTest.cs:281]: `result`

File: test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs [hunk 4]
```diff
@@ -0,0 +1,710 @@
...
+            }
+        }
+
+        // Same as PushCommand_PushToFileSystemSource, except that the directory is specified
+        // in UNC format.
+        [Fact]
+        public void PushCommand_PushToFileSystemSourceUncStyle()
+        {
+            // UNC only works in Windows. So skip this test if we're running on Unix,
+            if (Path.DirectorySeparatorChar == '/')
```
Comment (pranavkm) [inline test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs:94]: Environment.OsVersion

File: test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+                // Cleanup
+                Util.DeleteDirectory(packageDirectory);
+            }
+        }
+
+        // Asserts that the contents of two files are equal.
+        void AssertFileEqual(string fileName1, string fileName2)
+        {
+            byte[] content1, content2;
+            using (var r1 = new FileStream(fileName1, FileMode.Open))
```
Comment (pranavkm) [inline test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs:698]: Assert.Equal(File.ReadAllBytes(fileName1), File.ReadAllBytes(fileName2));

**Comment by pranavkm** [comment]
:shipit:

**Comment by emgarten** [comment]
:shipit:

File: test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs (hunk 4 above)
Comment (deepakaravindr) [inline test/NuGet.CommandLine.Test/NuGetPushCommandTest.cs:94]: https://github.com/NuGet/Home/issues/1053

File: test/NuGet.CommandLine.Test/NuGetListCommandTest.cs (hunk 2 above)
Comment (deepakaravindr) [inline test/NuGet.CommandLine.Test/NuGetListCommandTest.cs:1]: We don't seem to do this for the test files

**Pull request merged**